server/datastore/mysql/apple_mdm.go

```diff
@@ -2862,6 +2862,219 @@ func (ds *Datastore) GetMDMAppleDefaultSetupAssistant(ctx context.Context, teamI
 	return asst.ProfileUUID, asst.UploadedAt, nil
 }
 
+func (ds *Datastore) UpdateHostDEPAssignProfileResponses(ctx context.Context, payload *godep.ProfileResponse) error {
+	if payload == nil {
+		// caller should not ensure this does not happen
+		level.Debug(ds.logger).Log("msg", "update host dep assign profiles responses received nil payload")
+		return nil
+	}
+
+	success := make([]string, 0, len(payload.Devices))
+	notAccessible := make([]string, 0, len(payload.Devices))
+	failed := make([]string, 0, len(payload.Devices))
+	for serial, status := range payload.Devices {
+		switch strings.ToLower(status) {
+		case string(fleet.DEPAssignProfileResponseSuccess):
+			success = append(success, serial)
+		case string(fleet.DEPAssignProfileResponseNotAccessible):
```

Comment:
TODO: How should we handle not accessible responses? I've observed that unrecognized serial numbers get this response.

Reply:
thinking outloud, either:

1. we failed to remove a host after pinging the `/sync` (bug)
2. the host was removed from ABM but we didn't sync yet (race, edge case)
3. we saved the wrong serial number (bug, very unlikely?)
4. Apple error

should we log for now?

Reply:
@roperzh, logging for now makes sense to me. I think bugs related to this would generally manifest in the UI as a host that is stuck in MDM enroll status pending. So if we're logging the host serial as not accessible that would help future us to piece together the puzzle.

Have we seen any latency issues with the Apple service where we would need to worry that Fleet ingests a device via DEP sync but the device isn't yet accessible via the DEP assign service?